src/account_manager.rs

```diff
@@ -758,17 +763,27 @@ impl AccountManager {
     }
 
     /// Initialises the background polling and MQTT monitoring.
-    pub async fn start_background_sync(&mut self, polling_interval: Duration, automatic_output_consolidation: bool) {
+    pub async fn start_background_sync(
+        &self,
+        polling_interval: Duration,
+        automatic_output_consolidation: bool,
+    ) -> crate::Result<()> {
         Self::start_monitoring(self.accounts.clone()).await;
         let (stop_polling_sender, stop_polling_receiver) = broadcast_channel(1);
-        self.start_polling(polling_interval, stop_polling_receiver, automatic_output_consolidation);
-        self.stop_polling_sender.replace(stop_polling_sender);
+        self.start_polling(polling_interval, stop_polling_receiver, automatic_output_consolidation)?;
+        self.stop_polling_sender
+            .lock()
+            .map_err(|_| crate::Error::PoisonError)?
+            .replace(stop_polling_sender);
+        Ok(())
     }
 
     /// Stops the background polling and MQTT monitoring.
-    pub fn stop_background_sync(&mut self) {
-        if let Some(polling_handle) = self.polling_handle.take() {
+    pub fn stop_background_sync(&self) -> crate::Result<()> {
+        if let Some(polling_handle) = self.polling_handle.lock().unwrap().take() {
             self.stop_polling_sender
+                .lock()
+                .map_err(|_| crate::Error::PoisonError)?
```

Comment:
This is a bit inconsistent, you sometimes unwrap or map_err on the lock error. It's usually fine to always unwrap them.